### PR TITLE
[Keyboard]  Support MJ61 which is a 60% ANSI STD Hotswap RGB keyboard

### PR DESCRIPTION
**Pull request by melgeek001365**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/melgeek/mj61/mj61.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+};
+
+void matrix_init_kb(void) {
+    matrix_init_user();
+}
+
+void matrix_scan_kb(void) { matrix_scan_user(); }
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) { return process_record_user(keycode, record); }
+
```
Comment (drashna) [inline keyboards/melgeek/mj61/mj61.c:99]: If these aren't being used, then they can be removed. 
```suggestion
```

File: keyboards/melgeek/mj61/mj61.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
+void matrix_setup(void) {
+	palSetPadMode(GPIOA, 2, PAL_MODE_OUTPUT_PUSHPULL);
+	palSetPad(GPIOA, 2);
+}
```
Comment (drashna) [inline keyboards/melgeek/mj61/mj61.c:113]: Is there a specific need for this? 

Also:
```suggestion
void keyboard_pre_init_kb(void) {
    setPinOutput(A2);
    writePinHigh(A2);
    keyboard_pre_init_user();
}

File: keyboards/melgeek/mj61/readme.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# MelGeek MJ61
+
+![MJ61](https://drive.google.com/file/d/1siZCVPbHR-L7UVPa6R_-WPJDPoMtis9Q/view)
```
Comment (drashna) [inline keyboards/melgeek/mj61/readme.md:3]: limited access.

May want to host this on imgur or the like.

**Review by drashna — CHANGES_REQUESTED** [review]
Could you add a license header to all of the `c` and `h` files in this PR?

File: keyboards/melgeek/mj61/mj61.c (hunk 2 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj61/mj61.c:113]: > Is there a specific need for this?
> 
> Also:

Yes. GPIOA 2 is the power control of ISSI3741.

File: keyboards/melgeek/mj61/rev1/rules.mk [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/melgeek/mj61/rev1/rules.mk:18]:
```suggestion
BLUETOOTH_ENABLE = no          # Enable Bluetooth
```

File: keyboards/melgeek/mj61/rev1/config.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0xED61
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    MelGeek
+#define PRODUCT         MJ61
+#define DESCRIPTION     MJ61 ARM RGB keyboard
```
Comment (fauxpark) [inline keyboards/melgeek/mj61/rev1/config.h:9]:
```suggestion
```

File: keyboards/melgeek/mj61/readme.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+A hotswap 60% RGB Custom Mechanical keyboard.
+
+* Keyboard Maintainer: [melgeek](https://github.com/melgeek001365)
+* Hardware Supported: MJ61 V1.1.2 20200612
+* Hardware Availability: [MelGeek](https://www.melgeek.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make melgeek/mj61/rev1:default
+
```
Comment (fauxpark) [inline keyboards/melgeek/mj61/readme.md:14]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/mj61/readme.md (hunk 3 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj61/readme.md]: > limited access.
> 
> May want to host this on imgur or the like.

Done. Check it please

**Review by noroadsleft — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/melgeek/mj61/mj61.c [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mj61.h"
+
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
Comment (fauxpark) [inline keyboards/melgeek/mj61/mj61.c:19]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```

File: keyboards/melgeek/mj61/mj61.c [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A2);
+    writePinHigh(A2);
+    keyboard_pre_init_user();
+}
```
Comment (fauxpark) [inline keyboards/melgeek/mj61/mj61.c:122]:
```suggestion
}
#endif
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good.

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Comment by melgeek001365** [comment]
> Best Practices: Your Fork's Master: Update Often, Commit Never

Thank you for your advice!

File: keyboards/melgeek/mj61/mj61.c (hunk 8 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj61/mj61.c:123]: > 

Hey. Please check it again! Thank you very much

**Pull request merged**